infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/SecretAsContainerResourceProvisioner.java

```diff
@@ -0,0 +1,211 @@
+        if (targetContainerName != null && !container.getName().equals(targetContainerName)) {
+          continue;
+        }
+        for (Entry<String, String> entry : secret.getData().entrySet()) {
+          final String mountEnvName = envName(secret, entry.getKey());
+          container
+              .getEnv()
+              .add(
+                  new EnvVarBuilder()
+                      .withName(mountEnvName)
+                      .withValueFrom(
+                          new EnvVarSourceBuilder()
+                              .withSecretKeyRef(
+                                  new SecretKeySelectorBuilder()
+                                      .withName(secret.getMetadata().getName())
+                                      .withKey(entry.getKey())
+                                      .build())
+                              .build())
+                      .build());
+        }
+      }
+    }
+  }
+
+  private void mountAsFile(E env, Secret secret, String targetContainerName)
+      throws InfrastructureException {
+
+    final String mountPath = secret.getMetadata().getAnnotations().get(ANNOTATION_MOUNT_PATH);
+    if (mountPath == null) {
+      throw new InfrastructureException(
+          format(
+              "Unable to mount secret '%s': it has to be mounted as file, but mountPath is not specified."
+                  + "Please make sure you specified '%s' annotation of secret.",
+              secret.getMetadata().getName(), ANNOTATION_MOUNT_PATH));
+    }
+
+    Volume volumeFromSecret =
+        new VolumeBuilder()
+            .withName(secret.getMetadata().getName())
+            .withSecret(
+                new SecretVolumeSourceBuilder()
+                    .withNewSecretName(secret.getMetadata().getName())
+                    .build())
+            .build();
+
+    for (PodData podData : env.getPodsData().values()) {
+      if (!podData.getRole().equals(PodRole.DEPLOYMENT)) {
+        continue;
+      }
+      if (podData
+          .getSpec()
+          .getVolumes()
+          .stream()
+          .anyMatch(v -> v.getName().equals(volumeFromSecret.getName()))) {
+        throw new InfrastructureException(
+            format(
+                "Volume name '%s' provisioned from secret, clashes with existing volume name.",
+                volumeFromSecret.getName()));
+      }
+
+      podData.getSpec().getVolumes().add(volumeFromSecret);
+
+      for (Container container : podData.getSpec().getContainers()) {
+        if (targetContainerName != null && !container.getName().equals(targetContainerName)) {
+          continue;
+        }
+        if (container
+            .getVolumeMounts()
+            .stream()
+            .anyMatch(vm -> vm.getMountPath().equals(mountPath))) {
+          throw new InfrastructureException(
+              format(
+                  "Volume Mount path '%s' provisioned from secret, clashes with existing volume mount path.",
```

Comment:
The secret '%s' defines a mount path '%s' that clashes with another volume mount path already present on the workspace pod.